apache/nifi
# NIFI-11463 Allow IdentifyMimeType to use custom MIME types in addition to defaults

# Summary

[NIFI-11463](https://issues.apache.org/jira/browse/NIFI-11463)
This allows users to define their own mime types as well using the out of the box tika mime type definitions at the same time.

User defined mime types take precedence over the default definitions.

This also provides a workaround for [NIFI-11084](https://issues.apache.org/jira/browse/NIFI-11084) as noted in comment there.

To test,
For IdentifyMimeType Config File property, use  [.customConfig.xml](https://github.com/apache/nifi/blob/rel/nifi-1.23.2/nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/resources/TestIdentifyMimeType/.customConfig.xml) 

Test 1:
For Config Strategy use "add"
Pass through [jpg](https://github.com/apache/nifi/blob/rel/nifi-1.23.2/nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/resources/TestIdentifyMimeType/blueBtnBg.jpg) and [zip](https://github.com/apache/nifi/blob/rel/nifi-1.23.2/nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/resources/TestIdentifyMimeType/1.zip) files.
Verify mime.type is my/jpeg and application/zip, respectively

Test 2 (this verifies previously existing processor behaviour still works as before):
for Config Strategy use "replace"
pass through previous two files
Verify mime.type is my/jpeg and application/octet-stream, respectively.

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking


### Pull Request Tracking


### Pull Request Formatting


# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build


### Licensing


### Documentation
